### PR TITLE
Prompting for accounts (also bug fix: need mainnet account)

### PR DESCRIPTION
**Pull request by tallywiesenberg**
Adding account prompts to add/override accounts and endpoints (esp if the user does not have one/any)!

### Summary

Prompts user to pick/update chain_id, accounts, and endpoints, then sets up `TelliotCore` with these chosen configurations.
Also removes old `mainnet_config` variable in the blockhash generator!

### Steps Taken to QA Changes

Tested (seperately)...
- updating chain id
- updating endpoints
- updating accounts
- interfacing with the prompt for accounts' terminal menu

### Checklist

This pull request is:

- [ ] A documentation error, docs update, or typographical error fix
	- No tests or issue needed
- [x] A code fix
	- Please reference the related issue by including "Closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a bug report issue 
	- Please include tests. Fixes without tests will not be accepted unless it's related to the documentation only.
    - Please make sure docs are updated if need be
- [x] A feature implementation
	- Please reference the related issue by including "Closes `<link to issue>`" in this Pull Request's summary section. 
        - If no issue exists, please create a feature enhancement issue 
	- Please include tests
    - Please make sure docs updates are both thorough and easy to reproduce by somebody with limited knowledge of the feature that you are submitting

**Happy engineering!**

### CODE REVIEW

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks for your work on this so far @tallywiesenberg . Looks like:
- a few of the CI runs are failing
- need to update project dependencies if you're importing a new package
- if they select a new chain id, it should write that to the config file, so it uses it next time telliot is run. Same for any of the other configs.
- any config changes should be tested by reading the config files before and after
- except specific exceptions, clean up the comments
...
(: good progress

**Review by oraclown — CHANGES_REQUESTED** [review]
I think you just wanted me to comment on whether there's too much prompting. Just make it so there's as little steps to setup as possible while giving them the choice to update their configs.

- Print out all the current configs that are available (chain id, associated endpoint, & associated account.
- Ask once if they want to update anything.
- If yes (they want to update current configs), then ask update chain id, then account, then endpoint.
- If no, then proceed (and return configs). BUT, ask them to add endpoint and/or account if there's none associated with the current chain id (that they've said they're ok w/ using)

Something like:
```console
>>> telliot-feeds report
Current configuration:
Chain ID: 1
(it doesn't print any account configs here bc you don't happen to actually have any in this example, but it usually would print them before endpoint if there was one available)
Endpoint:
    - Node provider: blah
    - ...
...
Use these settings? (y/n): n (user types this)

Chain ID? (1): 80001 (user types this)
Account? (<-- prints that, or just says "Set up account:" if there's none available for the chain id they chose, then proceeds to ask them for the desired args of a chained account)  
(if there are accounts available for the chain id they chose, print numbered list of those accounts, user enters int to choose)
... After, something similar for Endpoint..
```

But as clean/pretty as possible. Have suggested defaults for each thing it prompts the user to enter.

**Comment by tallywiesenberg** [comment]
@oraclown  this is ready for review, and closes #343 . links to https://github.com/tellor-io/telliot-core/pull/320

**Comment by tallywiesenberg** [comment]
@akremstudy i corrected the account selection!

**Comment by akremstudy** [comment]
Thank you, @tallywiesenberg.  The account selection works.  But using the usual command ie ```telliot-feeds report -a my-account``` also forces you to go through the entire selection process, is that intended for now?

File: src/telliot_feeds/sources/blockhash_aggregator.py [hunk 1]
```diff
@@ -9,18 +9,19 @@
...
 from telliot_feeds.dtypes.datapoint import OptionalDataPoint
-from telliot_feeds.utils.cfg import mainnet_config
 from telliot_feeds.utils.log import get_logger
 
 logger = get_logger(__name__)
 
 
-cfg = mainnet_config()
+cfg = TelliotConfig()
+cfg.main.chain_id = 1
```
Comment (oraclown) [inline src/telliot_feeds/sources/blockhash_aggregator.py:24]: Can we add a logging message that chain id 1 is being set bc required when using the rng reporter?

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks for these updates @tallywiesenberg . So it looks like it's still asking if the user wants to change the chain id, endpoint, then finally account. See [here](https://github.com/tellor-io/telliot-feeds/pull/377#pullrequestreview-1111358810) again for the intended behavior. Print all configs out once, then ask to change each like you're doing. 

Another thing: Make sure your `except SpecificException`s or accept unknown ones and log them as `e`, do help debugging.

Also, go through and clean up some of the redundant comments in places where your code is already readable. For example, you don't need to comment "prompt for account name" if the code basically reads that way in English

Thanks again for these updates, we're getting closer (:

**Comment by oraclown** [comment]
> Thank you, @tallywiesenberg. The account selection works. But using the usual command ie `telliot-feeds report -a my-account` also forces you to go through the entire selection process, is that intended for now?

Yeah, basically. It should print all the config settings no matter what, and ask once if the user would like to update (any) of them. If not, proceed to whatever reporting..

File: src/telliot_feeds/sources/blockhash_aggregator.py (hunk 1 above)
Comment (tallywiesenberg) [reply src/telliot_feeds/sources/blockhash_aggregator.py:24]: sure!

**Review by oraclown — CHANGES_REQUESTED** [review]
Thanks @tallywiesenberg , feedback [here](https://www.loom.com/share/e01e2a2bfba24dc983b551a0095b8bbe)

- need to write config updates to the actual files and test this
- except specific exceptions
- ux is still a bit off
..more detail in the vid^

**Comment by tallywiesenberg** [comment]
One test `test_setup_account` does not pass on GH actions but passes locally with `tox`. I can troubleshoot it further or remove the test, thoughts @oraclown ? i can also replace the package if need be.

**Comment by tallywiesenberg** [comment]
also, added writing to config files, excepting specific exceptions, and requested updates to ux!

**Review by oraclown — APPROVED** [review]
Thanks for these changes @tallywiesenberg ! Nice tests too
I'm going to merge. Only thing I am a bit confused about is why we would let them continue progressing if certain configs weren't set. Wouldn't this just mean things will break once it gets to the reporter code?

**Pull request merged**